pkg/ccl/changefeedccl/changefeedbase/settings.go

```diff
@@ -239,7 +240,13 @@ var EventConsumerWorkers = settings.RegisterIntSetting(
 	settings.TenantWritable,
 	"changefeed.event_consumer_workers",
 	"the number of workers to use when processing events; 0 or 1 disables",
-	int64(util.ConstantWithMetamorphicTestRange("changefeed.consumer_max_workers", 8, 0, 32)),
+	func() int64 {
+		idealNumber := runtime.NumCPU() >> 2
+		if idealNumber < 1 {
+			return 1
+		}
+		return int64(idealNumber)
```

Comment:
Do we have prior art to using defaults that could change if e.g. cluster is upsized/moved?

I think we should change the semantics of this setting: -1 disables, 0 -- we pick reasonable default, anything >0 -- we use user provided value.
When we pick default, we should do this logic; but also, let's cap the max at 8.

Reply:
Just pushed updates. I moved the default value calculation to where we initialize the event consumer. This way, we can at least generate a new number whenever a changefeed is started/restarted.